pinot-core/src/main/java/org/apache/pinot/core/query/request/context/utils/BrokerRequestToQueryContextConverter.java

```diff
@@ -53,12 +55,81 @@ private BrokerRequestToQueryContextConverter() {
    * Converts the given {@link BrokerRequest} into a {@link QueryContext}.
    */
   public static QueryContext convert(BrokerRequest brokerRequest) {
-    return brokerRequest.getPinotQuery() != null ? convertSQL(brokerRequest) : convertPQL(brokerRequest);
+    if (brokerRequest.getPinotQuery() != null) {
+      QueryContext queryContext = convertSQL(brokerRequest.getPinotQuery(), brokerRequest);
+      queryContext.setGapfillType(GapfillUtils.getGapfillType(queryContext));
+      validateForGapfillQuery(queryContext);
```

Comment:
It seems like this validation is just checking syntax, can it be done on the Broker? If I am not mistaken almost all of Pinot's syntax checks happen on the Broker side.

Reply:
If I am not mistaken, BrokerRequestToQueryContextConverter is running inside Pinot Broker. @amrishlal can you clarify your comment further?

Reply:
From what I can see these syntax checks are happening on the Server side. Server receives an `InstanceRequest` from Broker in `InstanceRequestHandler.channelRead0` method and invokes the `ServerQueryRequest` constructor to create `ServerQueryRequest` object. The `ServerQueryRequest` constructor then calls makes the call:

    `_queryContext = BrokerRequestToQueryContextConverter.convert(brokerRequest);`

We should move the syntax checks as far up the stack as possible and reject bad queries as early as possible so that unnecessary processing can be avoided.

Reply:
Fixed